cmake/Modules/FindTinyXML2.cmake

```diff
@@ -6,10 +6,12 @@
 find_package(TinyXML2 CONFIG QUIET)
 if(TinyXML2_FOUND)
   message(STATUS "Found TinyXML2 via Config file: ${TinyXML2_DIR}")
-  if(NOT TINYXML2_LIBRARY AND TARGET tinyxml2)
+  if(NOT TINYXML2_LIBRARY)
     # in this case, we're probably using TinyXML2 version 5.0.0 or greater
     # in which case tinyxml2 is an exported target and we should use that
-    set(TINYXML2_LIBRARY tinyxml2)
+    if(TARGET tinyxml2 OR TARGET tinyxml2::tinyxml2)
+      set(TINYXML2_LIBRARY tinyxml2)
```

Comment:
I don't understand. If there is a library called `tinyxml2` then I would expect a library target called `tinyxml2`. If so, `if(TARGET tinyxml2)` would also be true, and the original code would work.

Reply:
Agreed, would it sometimes be `set(TINYXML2_LIBRARY tinyxml2::tinyxml2)`?

Reply:
I also don't understand. But removing ` OR TARGET tinyxml2::tinyxml2` results in this `set` not getting called and subsequent linker errors.

Before 43522c9, I had a separate conditional block for the namespaced target, which also works:

```cmake
if(TARGET tinyxml2::tinyxml2)
  set(TINYXML2_LIBRARY tinyxml2::tinyxml2)
endif()
```

But when we had an issue with this approach when applying it to Fast-RTPS, where the downstream `rmw_fastrtps_shared_cpp` package was complaining about an unresolvable target `tinyxml2::tinyxml2`... Removing the namespace from `TINYXML2_LIBRARY` resolved it. @Karsten1987 Do you have more insight as to why?

Reply:
The issue was resolved in Fast-RTPS by find_package'ing `TinyXML2` in it's cmake config.
I've updated this PR to reflect an earlier iteration.